### PR TITLE
Css class with dashes

### PR DESCRIPTION
**Pull request by ronachera**
NavigationHelper is now also able to render css classes with dashes. Previously each word of a class with dashes was seen as a class itself, f.e. "my-awesome-class" was rendered as three classes: my awesome class

[Note: I am happy to receive feedback on how to test css classes, right now I just made sure that the gsub works correctly, since a dash "-" is considered as a "non-word-character" and that's why it was 'gsubbed' before]

#69

### CODE REVIEW

**Comment by 5minpause** [comment]
> [Note: I am happy to receive feedback on how to test css classes, right now I just made sure that the gsub works correctly, since a dash "-" is considered as a "non-word-character" and that's why it was 'gsubbed' before]

This was a creative way and does test your changes, after all. 😃 

What you could change about it:
- `NavigationHelper` is a module so its methods can be tested.
- If you use a method that you call to make the change you need, you can test the method.

The problem you face here is that this change you implemented lives inside a huge method that needs to be refectored into several smaller pieces. Only then you can really test what happens there. If you are up to the task of refactoring that beast, go for it. Until then I would suggest you make a `# TODO` comment next to your changes and write a small comment for your test on why you test it like that. Just to make sure the next person working on this piece of code gets enough context to understand why things are like that right now.

Thanks very much for your changes and you PR!

File: app/helpers/goldencobra/navigation_helper.rb [hunk 1]
```diff
@@ -87,9 +88,9 @@ def navigation_menu(menue_id, options={})
         end
 
         if id_name.present?
-          result = content_tag(:ul, raw(content),id: "#{id_name}", class: "#{class_name} #{depth} navigation #{master_menue.css_class.to_s.gsub(/\W/,' ')}".squeeze(' ').strip)
+          result = content_tag(:ul, raw(content),id: "#{id_name}", class: "#{class_name} #{depth} navigation #{master_menue.css_class.to_s.gsub(/[^A-z\-]/,' ')}".squeeze(' ').strip)
```
Comment (marcometz) [inline app/helpers/goldencobra/navigation_helper.rb]: das \W läßt meiner Meinung nach auch Wörter, zu die eine Nummer enthalten was der neue Regexp nicht macht:
CSS Namen wie header1, header2 oder ähnliches würden jetzt also nicht mehr gehen.

ich würde `gsub(/[^A-z\-]/,' ')}` ergänzen um die Ziffern : `gsub(/[^A-z0-9\-]/,' ')}`

Comment (5minpause) [reply app/helpers/goldencobra/navigation_helper.rb]: Das ist richtig.

Comment (ronachera) [reply app/helpers/goldencobra/navigation_helper.rb]: oder gleich \w ? Ist ja ne whitelist. ( \w : Any word character (letter, number, underscore))

Comment (5minpause) [reply app/helpers/goldencobra/navigation_helper.rb]: Aber die `-` musst du dann wieder selbst ergänzen, oder?

Comment (marcometz) [reply app/helpers/goldencobra/navigation_helper.rb]: \w und zusätzlich \-  da wären dann auch "_" mit drin. Wäre für mich aber ok. (z.b.: `title_sub`)

**Pull request merged**